lib/system.ml

```diff
@@ -352,6 +352,57 @@ let fmt_transaction_result x =
   in
   str msg ++ fmt_duration duration ++ str msg2
 
+type instruction_count = (int, string) Result.t
+
+let instruction_count_add c1 c2 =
+  match (c1, c2) with
+  | (Ok i1  , Ok i2  ) -> Ok (i1 + i2)
+  | (Error _, _      ) -> c1
+  | (_      , Error _) -> c2
+
+type 'a instructions_result =
+  (('a * instruction_count), (Exninfo.iexn * instruction_count)) Result.t
+
+let count_instructions f x =
+  match Instr.reset () with
+  | exception Instr.Error m      ->
+      (* Error setting up the instruction counter, run [f] anyway. *)
+      begin
+        try Ok(f x, Error m) with e ->
+          Error(Exninfo.capture e, Error m)
+      end
+  | ()                             ->
```

Comment:
Please don't do this padding stuff
```suggestion
  | () ->
```
and same for all other alignment padding

Reply:
I think I took care of most of these.